### PR TITLE
JAMES-1780 Support accountId null in setVacationResponse

### PR DESCRIPTION
**Pull request by Laura-Royet**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java [hunk 1]
```diff
@@ -42,9 +42,13 @@ public static Builder builder() {
     public static class Builder {
 
         private Map<String, VacationResponse> update = Maps.newHashMap();
-
+
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java]: not needed

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java [hunk 2]
```diff
@@ -42,9 +42,13 @@ public static Builder builder() {
     public static class Builder {
 
         private Map<String, VacationResponse> update = Maps.newHashMap();
-
+ 
         public Builder accountId(String accountId) {
-            throw new NotImplementedException();
+            if (accountId != null) {
+                throw new NotImplementedException();
+            } else {
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java]: This else is not needed

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/SetVacationRequestTest.java [hunk 3]
```diff
@@ -51,4 +51,15 @@ public void accountIdIsNotImplemented() {
...
 
+    @Test
+    public void accountIdNullShouldBeConsideredAsNoAccountId() {
+        VacationResponse vacationResponse = VacationResponse.builder().id(VACATION_ID).textBody(Optional.of("any message")).build();
+        SetVacationRequest setVacationRequest = SetVacationRequest.builder()
+            .accountId(null)
+            .update(ImmutableMap.of(VACATION_ID, vacationResponse))
+            .build();
+
+        assertThat(setVacationRequest.getUpdate()).containsExactly(new AbstractMap.SimpleEntry<>(VACATION_ID, vacationResponse));
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/SetVacationRequestTest.java]: containsEntry

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java (hunk 1 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java]: Ok, removed.

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java (hunk 2 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/SetVacationRequest.java]: done.

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/SetVacationRequestTest.java (hunk 3 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/SetVacationRequestTest.java]: ok, changed.

**Comment by chibenwa** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by Laura-Royet** [comment]
test this please

**Pull request closed without merging**